### PR TITLE
RFC: one mdBook to rule them all 

### PR DESCRIPTION
**Pull request by matklad**
In this PR, I want to draft how we could centralize all our dev docs in a single, publicly accessible and publicly editable location. 

In this PR I want us to agree on:

* centralizing *all* dev docs in  single location 
* using [mdBook](https://rust-lang.github.io/mdBook/) as a specific medium for the documentation. 

To this end I:

* fleshed out the [landing page](https://matklad.github.io/nearcore/) of the book. Does it make sense to you?
* moved over some existing md docs: [style](https://matklad.github.io/nearcore/code/style.html) and [architecture](https://matklad.github.io/nearcore/architecture/index.html)
* moved one doc from confluence: [gc](https://matklad.github.io/nearcore/architecture/how/gc.html). 

See the rendered version here: https://matklad.github.io/nearcore/ 

Note the edit button as well: 

image

### CODE REVIEW

**Comment by akhi3030** [comment]
This is absolutely brilliant.  I am in full support for moving full steam ahead on this.

**Comment by mm-near** [comment]
how did you upload the images?

**Comment by matklad** [comment]
> This is absolutely brilliant. I am in full support for moving full steam ahead on this.

I don't think this is *absolutely* brilliant, there's certainly some drawbacks: 

* no lightweight comments
* no wiswig
* media management is worse than just drag'n'dropping images. 

> how did you upload the images?

Yeah, this is an important bit. For .svg illustrations, I put them in the media folder. I think we probably should try to stick to .svg as possible. 

For .png illustrations, it *would* be possble to put them in media, but I'd rather not. I think it's important to try to keep repo size reasonable. So I use my standard trick here: host images on github: https://github.com/matklad/nearcore/issues/2

And yeah, this is the policy I'd love to propose:

* default to .svg
* if there's a need to embed something more binary, host it "elsewhere", where the default "elsewhere" is a "media" issue in this repo

File: book/src/architecture/how/gc.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+![](https://user-images.githubusercontent.com/1711539/191359974-a002a8a9-7143-4a10-a506-5744426bb0af.png)
+
+Same as before, we’d remove up to 2 blocks in each run:
+
+![](https://user-images.githubusercontent.com/1711539/191359818-5ce228c4-a772-45ca-8161-a7ce9c2431b9.png)
+
+Until we catchup to the gc_stop.
+
+
+(the original drawings for this document are here: <https://docs.google.com/document/d/1BiEuJqm4phwQbi-fjzHMZPzDL-94z9Dqkc3XPNnxKJM/edit?usp=sharing>)
```
Comment (bowenwang1996) [inline book/src/architecture/how/gc.md:79]: This document isn't publicly accessible right? Should we remove the link?

**Review by bowenwang1996 — COMMENTED** [review]
This is epic! I like the name of the branch :)

File: book/src/architecture/how/gc.md (hunk 1 above)
Comment (matklad) [reply book/src/architecture/how/gc.md:79]: Yeah, we should do *something* here once we actually merge this stuff, I choose not to think about this yet. 


That being said, I suggest not worrying about leaving some links to private stuff:

* this should be low-effort-to-write kind of docs
* if link is broken for someone, they'll bug us

**Comment by matklad** [comment]
Chatted with marcin about images, the current solution is OK. So, I consider the RFC accepted and would move to implementation work, which is tracked at https://github.com/near/nearcore/issues/7670

**Pull request closed without merging**